doc/compiler/runtime/CodeCacheReclamation.md

```diff
@@ -0,0 +1,77 @@
+<!--
+Copyright (c) 2019, 2019 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Background
+
+The code generated by the compiler is stored in the Code Cache. It is allocated
+and managed by the compiler. However, by default, the compiler allocates the 
+total code repository during its initialization, and carves up smaller chunks of
+memory, namely Code Caches. Each Code Cache holds several methods as well as some
+space for Trampolines.
+
+Because the code repository is allocated at one time, there is a limited amount of
+space for methods. Additionally, code in these Code Caches can get stale or invalid
+for several reasons, such as recompilation, redefinition, class unloading, etc.
+Under these circumstances, the stale/invalid code takes up memory in the Code Cache 
+that could be used for something else.
+
+Code Cache Reclamation is the process by which unreachable code memory is reclaimed
+for reuse. The code generated for a method can either be reclaimed in a two-step 
+process, or all at once.
+
+# Two-Step Reclamation
+
+Code memory is reclaimed in two steps when the code is still reachable by executing 
```

Comment:
You may want to add that the two-step reclamation process is typically induced by recompilation.